### PR TITLE
Add new integration for vmware-esxi

### PR DESCRIPTION
**Pull request by vg-svitla**

### CODE REVIEW

File: VMWare/vmware-esxi/ingest/parser.yml [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+  set_additional_fields:
+    actions:
+      # Fields by definition that we get from messages
+      - set:
+          ecs.user.name: '{{parsed_event.result.username}}'
+          ecs.source.ip: '{{parsed_event.result.ip_address}}'
+          ecs.source.port: '{{parsed_event.result.port}}'
+          ecs.source.user.name: '{{parsed_event.result.source_username}}'
+          ecs.user_agent.original: '{{parsed_event.result.user_agent | trim}}'
```
Comment (squioc) [inline VMWare/vmware-esxi/ingest/parser.yml:155]: Same fields are already declared in ECS taxonomy. I think they are more suitable:
```suggestion
          user.name: '{{parsed_event.result.username}}'
          source.ip: '{{parsed_event.result.ip_address}}'
          source.port: '{{parsed_event.result.port}}'
          source.user.name: '{{parsed_event.result.source_username}}'
          user_agent.original: '{{parsed_event.result.user_agent | trim}}'
```

File: VMWare/vmware-esxi/ingest/parser.yml [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+          file.name: '{{parsed_event.result.filename}}'
+          wmware.esxi.key.id: '{{parsed_event.result.key_id | trim}}'
+          wmware.esxi.event.serial_number: '{{parsed_event.result.serial_number}}'
+          host.name: '{{parsed_event.result.server_name}}'
+          event.outcome: '{{parsed_event.result.outcome}}'
+
+      # It is necessary to postprocess the command line because it might contain `'` at the beginning and/or at the end
+      - set:
+          ecs.process.command_line: >
```
Comment (squioc) [inline VMWare/vmware-esxi/ingest/parser.yml:164]:
```suggestion
          process.command_line: >
```

File: VMWare/vmware-esxi/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+           {{ parsed_event.result.command[1:] }}
+           {%- elif parsed_event.result.command.endswith("'"): -%}
+           {{ parsed_event.result.command[:-1] }}
+           {%- else -%}
+           {{ parsed_event.result.command }}
+           {%- endif -%}
+
+      # We should check not only for pid but also for process_keyword because some logs have pid but we don't want to set it in output
+      - set:
+          ecs.process.pid: '{{parsed_event.result.pid}}'
```
Comment (squioc) [inline VMWare/vmware-esxi/ingest/parser.yml:177]:
```suggestion
          process.pid: '{{parsed_event.result.pid}}'
```

File: VMWare/vmware-esxi/_meta/fields.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+ecs.process.pid:
+  name: ecs.process.pid
+  description: "Process PID"
+  type: number
+
+ecs.source.port:
+  name: ecs.source.port
+  description: "Port number"
+  type: number
```
Comment (squioc) [inline VMWare/vmware-esxi/_meta/fields.yml:34]: Same fields exist in the ECS taxomony

File: VMWare/vmware-esxi/_meta/fields.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+host.name:
+  name: host.name
+  description: "Host name"
+  type: keyword
```
Comment (squioc) [inline VMWare/vmware-esxi/_meta/fields.yml:39]: Not already existing in ECS?

File: VMWare/vmware-esxi/tests/AUTH_CONNECTION_disconnected.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      "vendor": "VMware",
+      "product": "ESXi"
+    },
+    "event": {
+      "kind": "event",
+      "category": [
+        "authentication"
+      ],
+      "type": [
+        "info"
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/AUTH_CONNECTION_disconnected.json:23]: For this event, a more suitable `event.type` will be `["end"]` as it represent a disconnection.

File: VMWare/vmware-esxi/tests/AUTH_CONNECTION_event.json [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+      "vendor": "VMware",
+      "product": "ESXi"
+    },
+    "event": {
+      "kind": "event",
+      "category": [
+        "authentication"
+      ],
+      "type": [
+        "info"
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/AUTH_CONNECTION_event.json:23]: For this event, a more suitable `event.type` will be `["end"]` as it represent a disconnection.

File: VMWare/vmware-esxi/tests/AUTH_CONNECTION_recieved_disconnect.json [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+      "vendor": "VMware",
+      "product": "ESXi"
+    },
+    "event": {
+      "kind": "event",
+      "category": [
+        "authentication"
+      ],
+      "type": [
+        "info"
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/AUTH_CONNECTION_recieved_disconnect.json:23]: For this event, a more suitable `event.type` will be `["end"]` as it represent a disconnection.

File: VMWare/vmware-esxi/tests/AUTH_PUBLIC_KEY.json [hunk 9]
```diff
@@ -0,0 +1,37 @@
+{
+  "input": {
+    "sekoiaio": {
+      "intake": {
+        "dialect": "vmware-esxi",
+        "dialect_uuid": "2b13307b-7439-4973-900a-2b58303cac90"
+      }
+    },
+    "message": "Accepted publickey for root from 1.2.3.4 port 33398 ssh2: RSA SHA256:qzkqJPyBUdiJUC4i/wbJzkYLTcUWwXRArUT90bUw2E0"
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/AUTH_PUBLIC_KEY.json:9]: Maybe we can extract the publickey in a custom field with the hash of the key and the algorithm of the hash.

File: VMWare/vmware-esxi/tests/AUTH_SESSION_closed.json [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+      "vendor": "VMware",
+      "product": "ESXi"
+    },
+    "event": {
+      "kind": "event",
+      "category": [
+        "session"
+      ],
+      "type": [
+        "info"
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/AUTH_SESSION_closed.json:23]: For this event, a more suitable `event.type` will be `["end"]` as it represent the end of a session.

File: VMWare/vmware-esxi/tests/AUTH_SESSION_closed.json [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    "sekoiaio": {
+      "intake": {
+        "dialect": "vmware-esxi",
+        "dialect_uuid": "2b13307b-7439-4973-900a-2b58303cac90"
+      }
+    },
+    "message": "pam_unix(sshd:session): session closed for user root"
+  },
+  "expected": {
+    "message": "pam_unix(sshd:session): session opened for user root",
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/AUTH_SESSION_closed.json:12]:
```suggestion
    "message": "pam_unix(sshd:session): session closed for user root",
```

File: VMWare/vmware-esxi/tests/AUTH_SESSION_opened.json [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+      "vendor": "VMware",
+      "product": "ESXi"
+    },
+    "event": {
+      "kind": "event",
+      "category": [
+        "session"
+      ],
+      "type": [
+        "info"
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/AUTH_SESSION_opened.json:23]: For this event, a more suitable `event.type` will be `["start"]` as it represent the start of a session.

File: VMWare/vmware-esxi/tests/AUTH_SSH_PASSWORD.json [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+    "observer": {
+      "vendor": "VMware",
+      "product": "ESXi"
+    },
+    "event": {
+      "kind": "event",
+      "category": ["process", "session", "authentication"],
+      "type": [
+        "info"
+      ]
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/AUTH_SSH_PASSWORD.json:22]: I think that `event.category` = `configuration` and `event.type` = `change` will be more suitable.

File: VMWare/vmware-esxi/tests/AUTH_SSH_event_closed.json [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+      "vendor": "VMware",
+      "product": "ESXi"
+    },
+    "event": {
+      "kind": "event",
+      "category": [
+        "session"
+      ],
+      "type": [
+        "info"
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/AUTH_SSH_event_closed.json:23]: For this event, a more suitable `event.type` will be `["end"]` as it represent a disconnection.

File: VMWare/vmware-esxi/tests/AUTH_SSH_event_opened.json [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+      "vendor": "VMware",
+      "product": "ESXi"
+    },
+    "event": {
+      "kind": "event",
+      "category": [
+        "session"
+      ],
+      "type": [
+        "info"
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/AUTH_SSH_event_opened.json:23]: For this event, a more suitable `event.type` will be `["start"]` as it represent the starting of a new session.

File: VMWare/vmware-esxi/tests/AUTH_SSH_process_closed.json [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+      "product": "ESXi"
+    },
+    "event": {
+      "kind": "event",
+      "category": [
+          "process",
+          "session"
+      ],
+      "type": [
+        "error"
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/AUTH_SSH_process_closed.json:24]: `error` for `event.type` seems only suitable for category `web` and `database` in the ECS specification.
I think a more suitable `event.type` will be `["end"]`, as it represent the end of the session and put the log level in `log.level`

File: VMWare/vmware-esxi/tests/AUTH_SSH_process_opened.json [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+      "product": "ESXi"
+    },
+    "event": {
+      "kind": "event",
+      "category": [
+        "process",
+        "session"
+      ],
+      "type": [
+        "info"
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/AUTH_SSH_process_opened.json:24]: For this event, a more suitable event.type will be ["start"] as it represent the starting of a new session.

File: VMWare/vmware-esxi/tests/NETWORK_HOSTNAME.json [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+  "expected": {
+    "message": " DHCPDISCOVER on vmk1 to 255.255.255.255 port 67 interval 1",
+    "observer": {
+      "vendor": "VMware",
+      "product": "ESXi"
+    },
+    "event": {
+      "kind": "event",
+      "category": [
+        "process"
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/NETWORK_HOSTNAME.json:20]: I think `network` will be a more suitable category

File: VMWare/vmware-esxi/tests/NETWORK_SOURCE.json [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+  "expected": {
+    "message": "Connection from 1.2.3.4 port 33398",
+    "observer": {
+      "vendor": "VMware",
+      "product": "ESXi"
+    },
+    "event": {
+      "kind": "event",
+      "category": [
+        "connection"
```
Comment (squioc) [inline VMWare/vmware-esxi/tests/NETWORK_SOURCE.json:20]: Category `connection` doesn't exist. [network](https://www.elastic.co/guide/en/ecs/8.6/ecs-allowed-values-event-category.html#ecs-event-category-network) is more suitable.

File: VMWare/_meta/manifest.yml [hunk 20]
```diff
@@ -0,0 +1,5 @@
+uuid: 32131dce-2899-4271-997d-5c6d1f79360d
+name: VMWare
+slug: "vmware"
+description: >-
+  This module contains necessary parser configurations to work with VMWare logs format
```
Comment (squioc) [inline VMWare/_meta/manifest.yml:5]:
```suggestion
  VMware is a virtualization technology company. It's offering software hypervisors serving virtual computers, virtual storages or virtual networks.
```

File: VMWare/vmware-esxi/_meta/manifest.yml [hunk 21]
```diff
@@ -0,0 +1,8 @@
+uuid: 2b13307b-7439-4973-900a-2b58303cac90
+name: vmware-esxi
+slug: vmware-esxi
+
+description: >-
+  The description of the intake
```
Comment (squioc) [inline VMWare/vmware-esxi/_meta/manifest.yml:6]:
```suggestion
  VMware ESXi is a hypervisor and an operation system. It serves virtual computers while running directly on the server hardware
```

File: VMWare/vmware-esxi/_meta/manifest.yml [hunk 22]
```diff
@@ -0,0 +1,8 @@
...
+
+data_sources:
```
Comment (squioc) [inline VMWare/vmware-esxi/_meta/manifest.yml:8]: Can you add the most suitable datasources from [this list](https://github.com/SEKOIA-IO/intake-formats/blob/main/doc/datasources.md)?

**Review by squioc — CHANGES_REQUESTED** [review]
Good work.

Some changes to do to field names and categorization of the events.

File: VMWare/vmware-esxi/ingest/parser.yml (hunk 2 above)
Comment (vg-svitla) [reply VMWare/vmware-esxi/ingest/parser.yml:164]: Make sense but should we then put then also `ecs.process.pid` to `process.pid`?

File: VMWare/vmware-esxi/ingest/parser.yml (hunk 3 above)
Comment (vg-svitla) [reply VMWare/vmware-esxi/ingest/parser.yml]: Thank you! Fixed

File: VMWare/vmware-esxi/ingest/parser.yml (hunk 2 above)
Comment (vg-svitla) [reply VMWare/vmware-esxi/ingest/parser.yml]: Thank you! Fixed

**Review by squioc — APPROVED** [review]

**Pull request merged**